[2022-12-11T14:49:59.160+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: ArtistRevenue.ReadData scheduled__2022-12-10T00:00:00+00:00 [queued]>
[2022-12-11T14:49:59.170+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: ArtistRevenue.ReadData scheduled__2022-12-10T00:00:00+00:00 [queued]>
[2022-12-11T14:49:59.171+0000] {taskinstance.py:1283} INFO - 
--------------------------------------------------------------------------------
[2022-12-11T14:49:59.172+0000] {taskinstance.py:1284} INFO - Starting attempt 4 of 9
[2022-12-11T14:49:59.173+0000] {taskinstance.py:1285} INFO - 
--------------------------------------------------------------------------------
[2022-12-11T14:49:59.184+0000] {taskinstance.py:1304} INFO - Executing <Task(PythonOperator): ReadData> on 2022-12-10 00:00:00+00:00
[2022-12-11T14:49:59.192+0000] {standard_task_runner.py:55} INFO - Started process 9869 to run task
[2022-12-11T14:49:59.210+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'ArtistRevenue', 'ReadData', 'scheduled__2022-12-10T00:00:00+00:00', '--job-id', '216', '--raw', '--subdir', 'DAGS_FOLDER/artist_revenue.py', '--cfg-path', '/tmp/tmpqz25j2ih']
[2022-12-11T14:49:59.229+0000] {standard_task_runner.py:83} INFO - Job 216: Subtask ReadData
[2022-12-11T14:49:59.318+0000] {task_command.py:389} INFO - Running <TaskInstance: ArtistRevenue.ReadData scheduled__2022-12-10T00:00:00+00:00 [running]> on host 83e68f36074a
[2022-12-11T14:49:59.384+0000] {taskinstance.py:1513} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=Gama Ariefsadya
AIRFLOW_CTX_DAG_ID=ArtistRevenue
AIRFLOW_CTX_TASK_ID=ReadData
AIRFLOW_CTX_EXECUTION_DATE=2022-12-10T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=4
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-12-10T00:00:00+00:00
[2022-12-11T14:49:59.403+0000] {logging_mixin.py:137} INFO - {'artist_id': 1, 'artist_name': 'AC/DC', 'album_id': 1, 'album_title': 'For Those About To Rock We Salute You', 'track_id': 6, 'track_name': 'Put The Finger On You', 'invoice_line_id': 3, 'invoice_id': 3, 'invoice_date': datetime.date(2009, 1, 3), 'total': Decimal('0.99')}
[2022-12-11T14:49:59.404+0000] {logging_mixin.py:137} INFO - {'artist_id': 1, 'artist_name': 'AC/DC', 'album_id': 1, 'album_title': 'For Those About To Rock We Salute You', 'track_id': 8, 'track_name': 'Inject The Venom', 'invoice_line_id': 4, 'invoice_id': 4, 'invoice_date': datetime.date(2009, 1, 6), 'total': Decimal('1.98')}
[2022-12-11T14:49:59.404+0000] {logging_mixin.py:137} INFO - {'artist_id': 1, 'artist_name': 'AC/DC', 'album_id': 1, 'album_title': 'For Those About To Rock We Salute You', 'track_id': 10, 'track_name': 'Evil Walks', 'invoice_line_id': 5, 'invoice_id': 5, 'invoice_date': datetime.date(2009, 1, 11), 'total': Decimal('0.99')}
[2022-12-11T14:49:59.404+0000] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/artist_revenue.py", line 46, in ReadData
    revenues=json.dumps(revenues)
  File "/usr/local/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/usr/local/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/local/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type date is not JSON serializable
[2022-12-11T14:49:59.414+0000] {taskinstance.py:1327} INFO - Marking task as UP_FOR_RETRY. dag_id=ArtistRevenue, task_id=ReadData, execution_date=20221210T000000, start_date=20221211T144959, end_date=20221211T144959
[2022-12-11T14:49:59.422+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 216 for task ReadData (Object of type date is not JSON serializable; 9869)
[2022-12-11T14:49:59.476+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2022-12-11T14:49:59.501+0000] {taskinstance.py:2582} INFO - 0 downstream tasks scheduled from follow-on schedule check
